spring-cloud-stream-reactive/src/test/java/org/springframework/cloud/stream/reactive/StreamListenerReactorTests.java

```diff
@@ -0,0 +1,272 @@
+		processor.input().send(MessageBuilder.withPayload("fail").setHeader("contentType", "text/plain").build());
+	}
+
+	@Test
+	public void testInputOutputArgsWithMessage() throws Exception {
+		ConfigurableApplicationContext context = SpringApplication.run(TestInputOutputArgsWithMessage.class,
+				"--server.port=0");
+		sendMessageAndValidate(context);
+		context.close();
+	}
+
+	@Test
+	public void testInputOutputArgsWithFluxSender() throws Exception {
+		ConfigurableApplicationContext context = SpringApplication.run(TestInputOutputArgsWithFluxSender.class,
+				"--server.port=0");
+		// send multiple message
+		sendMessageAndValidate(context);
+		sendMessageAndValidate(context);
+		sendMessageAndValidate(context);
+		context.close();
+	}
+
+	@Test
+	public void testInputOutputArgsWithFluxSenderAndFailure() throws Exception {
+		ConfigurableApplicationContext context = SpringApplication.run(TestInputOutputArgsWithFluxSenderAndFailure.class, "--server.port=0");
+		sendMessageAndValidate(context);
+		sendFailingMessage(context);
+		sendMessageAndValidate(context);
+		context.close();
+	}
+
+	@Test
+	public void testReturn() throws Exception {
+		ConfigurableApplicationContext context = SpringApplication.run(TestReturn.class, "--server.port=0");
+		sendMessageAndValidate(context);
+		sendMessageAndValidate(context);
+		sendMessageAndValidate(context);
+		context.close();
+	}
+
+	@Test
+	public void testReturnWithFailure() throws Exception {
+		ConfigurableApplicationContext context = SpringApplication.run(TestReturnWithFailure.class, "--server.port=0");
+		sendMessageAndValidate(context);
+		sendFailingMessage(context);
+		sendMessageAndValidate(context);
+		context.close();
+	}
+
+	@Test
+	public void testReturnWithMessage() throws Exception {
+		ConfigurableApplicationContext context = SpringApplication.run(TestReturnWithMessage.class, "--server.port=0");
+		sendMessageAndValidate(context);
+		context.close();
+	}
+
+	@Test
+	public void testReturnWithPojo() throws Exception {
+		ConfigurableApplicationContext context = SpringApplication.run(TestReturnWithPojo.class, "--server.port=0");
+		@SuppressWarnings("unchecked")
+		Processor processor = context.getBean(Processor.class);
+		processor.input().send(MessageBuilder.withPayload("{\"message\":\"helloPojo\"}")
+				.setHeader("contentType", "application/json").build());
+		MessageCollector messageCollector = context.getBean(MessageCollector.class);
+		Message<?> result = messageCollector.forChannel(processor.output()).poll(1000, TimeUnit.MILLISECONDS);
+		assertThat(result).isNotNull();
+		assertThat(result.getPayload()).isInstanceOf(BarPojo.class);
+		assertThat(((BarPojo) result.getPayload()).getBarMessage()).isEqualTo("helloPojo");
+		context.close();
+	}
+
+
+	@EnableBinding(Processor.class)
+	@EnableAutoConfiguration
+	public static class TestInputOutputArgs {
+
+		@StreamListener
+		public void receive(@Input("input") Flux<String> input, @Output("output") FluxSender output) {
```

Comment:
Might be good to use `Processor.INPUT` and `Processor.OUTPUT` to refer the message channel names`input` and `output`. If agreed, the same can be applied in all the test methods.